wordpress-objects/Clarkson_Post_Type.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Clarkson Post Type.
+ *
+ * @package CLARKSON\Objects
+ */
+
+/**
+ * Clarkson Post Type class.
+ */
+class Clarkson_Post_Type {
+
+	/**
+	 * @var \WP_Post_Type
+	 */
+	protected $_post_type;
+
+	/**
+	 * @var array
+	 */
+	protected static $post_types;
+
+	public function __construct( \WP_Post_Type $post_type ) {
+		$this->_post_type = $post_type;
+	}
+
+	/**
+	 * Get post type object by post type name
+	 */
+	public static function get( string $post_type ): Clarkson_Post_Type {
+		$class            = get_called_class();
+		$post_type_object = get_post_type_object( $post_type );
+		return new $class( $post_type_object );
+	}
+
+	/**
+	 * Check if an existing field is requested.
+	 *
+	 * @param string $name Field to search by.
+	 *
+	 * @throws Exception Error message.
+	 */
+	public function __get( string $name ) {
+		if ( in_array( $name, array(), true ) ) {
```

Comment:
Yeah it will now throw an exception always when accessing direct properties. Is still from when this was an archive object because wordpress hasn't got that and i didn't change it except for the comment lol